controllers/sync/policy_spec_sync.go

```diff
@@ -93,14 +96,16 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 	}
 
 	managedPlc := &policiesv1.Policy{}
-	err = r.ManagedClient.Get(ctx, request.NamespacedName, managedPlc)
+	err = r.ManagedClient.Get(ctx, types.NamespacedName{Namespace: r.TargetNamespace, Name: request.Name}, managedPlc)
 
 	if err != nil {
 		if errors.IsNotFound(err) {
 			// not found on managed cluster, create it
 			reqLogger.Info("Policy not found on managed cluster, creating it...")
 
 			managedPlc = instance.DeepCopy()
+			managedPlc.Namespace = r.TargetNamespace
+			managedPlc.Labels["policy.open-cluster-management.io/cluster-namespace"] = r.TargetNamespace
```

Comment:
I just noticed that this label is set on the replicated policy so I was correcting the value.

Reply:
Ah, curious. Yeah, I see it in the propagator now.

Reply:
I think I'm also getting my wires crossed about what policy is going where and where everything is happening. Have you dug into what this label is even used for or what it's supposed to mean? I'm wondering if maybe we don't need to be updating it at all in this instance.

Reply:
@dhaiducek it seems that template sync will sync these labels to the policy templates and some controllers (IAM and Cert) use those labels to get the namespace in the `ExtractNamespaceLabel` rather than getting the namespace directly from the object. I don't know why that's the case, but I'd rather keep this in to not break that.

Reply:
Okay. If only we had a common policy framework... 🤔 😉

Reply:
> Okay. If only we had a common policy controller framework... thinking wink

/me nags @JustinKuli :smile:

Reply:
If only...